tests/unit/s2n_handshake_test.c

```diff
@@ -188,6 +188,9 @@ int test_cipher_preferences(struct s2n_config *server_config, struct s2n_config
             GUARD(try_handshake(server_conn, client_conn));
 
             EXPECT_STRING_EQUAL(s2n_connection_get_cipher(server_conn), expected_cipher->name);
+            uint8_t iana_value[2] = { 0, 0 };
+            EXPECT_SUCCESS(s2n_connection_get_cipher_iana_value(server_conn, &iana_value[0], &iana_value[1]));
+            EXPECT_EQUAL(memcmp(expected_cipher->iana_value, iana_value, sizeof(iana_value)), 0);
```

Comment:
It might be better to write a dedicated test for this API rather than expanding this one, which is already quite large and complicated. You could still test it for every cipher suite.

Reply:
I considered that but wondered how good of a test it would be to force the connection's negotiated cipher suite to a particular value and compare the API to the internal value. It seemed like it would just be mostly copy-paste from this test.

Reply:
I'd still say the copy-paste would be worth it. Adding every new API to the same single unit test doesn't seem sustainable :\ And since the stated purpose of the test is "Verify that a handshake succeeds for every available cipher in the default list" I'd be worried about refactors erasing / changing your less explicit test.